### PR TITLE
lock: Allow to create and select multiple locks CATW-1246

### PR DESCRIPTION
**Pull request by tloisNeo9**

### CODE REVIEW

**Review by BenjD90 — CHANGES_REQUESTED** [review]
Avoid to create a memory leak by feeling a map with new lock names, prefer to use a suffix to add to the acquire and release methods.

File: test/locks.test.ts [hunk 1]
```diff
@@ -0,0 +1,588 @@
+import { N9Log } from '@neo9/n9-node-log';
+import { waitFor } from '@neo9/n9-node-utils';
+import ava, { Assertions } from 'ava';
+import { Logger } from 'mongodb';
```
Comment (BenjD90) [inline test/locks.test.ts:4]: not used

File: test/locks.test.ts [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+import { init } from './fixtures/utils';
+
+export class TestItem extends BaseMongoObject {
+	public key: string;
+	public i?: number;
+}
+
+const codeRegexp = new RegExp(/^[0-9a-f]{32}$/);
+const col = 'locks';
+const threeSecs = 3_000;
```
Comment (BenjD90) [inline test/locks.test.ts:16]:
```suggestion
const threeSecsInMs = 3_000;
```

File: src/lock.ts [hunk 3]
```diff
@@ -85,17 +130,40 @@ export class N9MongoLock {
...
-		const ret: boolean = await new Promise<boolean>((resolve, reject) => {
-			this.lock.release(code, (err: any, ok: boolean) => {
-				if (err) return reject(err);
-				return resolve(ok);
-			});
-		});
-		// Wait to let enough time to someone else to pick the lock
-		await waitFor(this.options.n9MongoLockOptions.waitDurationMsMax + 5);
+	public async release(code: string, suffix?: string): Promise<boolean> {
+		const now = Date.now();
```
Comment (BenjD90) [inline src/lock.ts:137]: should be moved to a private release function to be more clear

File: src/lock.ts [hunk 4]
```diff
@@ -1,16 +1,14 @@
 import { N9Error, waitFor } from '@neo9/n9-node-utils';
+import * as crypto from 'crypto';
 import * as _ from 'lodash';
 import * as mongodb from 'mongodb';
-import * as mongoDbLock from 'mongodb-lock';
+import { LangUtils } from '.';
```
Comment (BenjD90) [inline src/lock.ts:5]:
```suggestion
import { LangUtils } from './lang-utils';
```
avoid loops in imports

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/lock.ts (hunk 3 above)
Comment (tloisNeo9) [reply src/lock.ts:137]: Don't we need to access this function from other places ?

**Pull request merged**